packages/ibm-products/src/components/Datagrid/Datagrid/addons/CustomizeColumns/DraggableItemsList.js

```diff
@@ -40,16 +40,14 @@ export const DraggableItemsList = ({
   const draggableClass = `${blockClass}__draggable-item`;
   const visibleCols = columns
     // hide the columns without Header, e.g the sticky actions, spacer
-    .filter((colDef) => {
-      return !!getColTitle(colDef);
-    })
+    .filter((colDef) => getNodeTextContent(colDef.Header).trim().length !== 0)
     .filter(Boolean)
     .filter((colDef) => !colDef.isAction)
+    .filter((colDef) => colDef.id !== 'spacer')
     .filter((colDef) => {
       return (
         filterString.length === 0 ||
-        (getColTitle(colDef)?.toLowerCase().includes(filterString) &&
-          colDef.id !== 'spacer')
+        getNodeTextContent(colDef.Header)?.toLowerCase().includes(filterString)
```

Comment:
Just a nit. `getNodeTextContent()` always returns a string, so optional chaining to test if the result is a string shouldn't be necessary.